docs/devguide/xtypes.rst

```diff
@@ -429,15 +460,49 @@ Type Consistency Enforcement
 ..
     Sect<16.5>
 
+.. _xtypes--typeconsistencyenforcementqospolicy:
+
+TypeConsistencyEnforcementQosPolicy
+===================================
+
 The Type Consistency Enforcement QoS policy lets the application fine-tune details of how types may differ between writers and readers.
-The policy is only applies to data readers.
+The policy only applies to data readers.
 This means that each reader can set its own policy for how its type may vary from the types of the writers that it may match.
 
 There are six members of the ``TypeConsistencyEnforcementQosPolicy`` struct defined by XTypes, but OpenDDS only supports setting one of them: ``ignore_member_names``.
 All other members should be kept at their default values.
 
-``ignore_member_names`` defaults to ``FALSE`` so member names (along with Member IDs, see :ref:`xtypes--member-id-assignment`) are significant for type compatibility.
-Changing this to TRUE means that only Member IDs are used for type compatibility.
+``ignore_member_names`` defaults to ``FALSE`` so member names (along with member IDs, see :ref:`xtypes--member-id-assignment`) are significant for type compatibility.
+Changing this to ``TRUE`` means that only member IDs are used for type compatibility.
+
+.. _xtypes--type-compatibility:
+
+Type Compatibility
+==================
+
+When a reader/writer match is happening, type consistency enforcement checks that the two types are compatible according to the type objects if they are available.
+This check will not happen if OpenDDS has been :ref:`configured not to generate or use type objects <xtypes--representing-types-with-typeobject-and-dynamictype>` or if the remote DDS doesn't support type objects.
+The full type object compatibility check is too detailed to reproduce here.
+It can be found in section 7.2.4 of the XTypes 1.3 specification.
+In general though two topic types and their nested types are compatible if:
+
+  * Extensibilities of shared types match
+  * Extensibility rules haven't been broken, for example:
+
+    * Changing a ``@final`` struct
+    * Adding a member in the middle of an ``@appendable`` struct
+
+  * Length bounds of strings and sequences are the same or greater
+  * Lengths of arrays are exactly the same
+  * The keys of the types match exactly
+  * Shared member IDs match when required, like when they are final or are being used as keys
+
+If the type objects are compatible then the match goes ahead.
+If one or both type objects are not available, then OpenDDS falls back to checking if the topic type names match.
```

Comment:
It might be more clear to call this type "TypeSupport type name" although they're both correct.